### ISSUE TITLE
Large XML Files crash language server

### ISSUE THREAD

**Issue by lunarwtr** [redhat-developer/vscode-xml#696]
I would like to see the vscode extension have a configuration to prevent loading & subsequently crashing due to large xml files.  If a file is greater than a defaulted size.. don't try to send to language server and show a message perhaps show a vscode notification as to why.

**Comment by angelozerr** [comment]
> If a file is greater than a defaulted size.. don't try to send to language server and show a message perhaps show a vscode notification as to why.

This feature should be part in vscode-languageclient  https://github.com/microsoft/vscode-languageserver-node I think.

**Comment by angelozerr** [comment]
@lunarwtr could you share your big XML please.

**Comment by lunarwtr** [comment]
I don't expect it to provide language support on a file so big.. I'd expect language server to bail out since a limit has been reached.  Instead it crashes and won't relaunch until vscode restarted.

**Comment by angelozerr** [comment]
> I don't expect it to provide language support on a file so big.. I'd expect language server to bail out since a limit has been reached. Instead it crashes and won't relaunch until vscode restarted.

I can open it without crash and outline is filled too. Are you editing (type some content) this file to crash the language server?

**Comment by lunarwtr** [comment]
crashes immediately on loading file.  Ensure you haven't' customized language server settings (i.e. greater memory etc.)  Just default settings upon extension install.

![image](https://user-images.githubusercontent.com/9368775/167444593-29d8e5ff-4bd2-47d2-a443-41614d14dd2f.png)

**Comment by fbricon** [comment]
Yeah I see the same JVM crash, when Lemminx runs in the JVM. If you set `"xml.server.preferBinary": true` in your settings.json, the binary will start, complain about outline limited to 5000 items, but completion/validation will work, albeit pretty slowly.
